src/library_syscall.js

```diff
@@ -451,10 +451,9 @@ var SyscallsLibrary = {
     var name = {{{ makeGetValue('message', C_STRUCTS.msghdr.msg_name, '*') }}};
     var namelen = {{{ makeGetValue('message', C_STRUCTS.msghdr.msg_namelen, 'i32') }}};
     if (name) {
-      var info = readSockaddr(name, namelen);
-      if (info.errno) return -info.errno;
```

Comment:
Looks like `getSocketAddress` will throw if there is an errno?

https://github.com/emscripten-core/emscripten/blob/1dd13b720a8b550d65e4fcb68fe41281b0d7ba7a/src/library_syscall.js#L330

Is that difference not noticeable?

Reply:
Oh, I guess we have those catches that convert, so it would be.

Reply:
Yes, all these syscalls are wrapped in try/catch to catch ErrnoError.